ocaml/forkexecd/test/fe_test.ml

```diff
@@ -187,6 +198,44 @@ let test_input () =
   expect input out ;
   print_endline "Completed input tests"
 
+(* This test tests a failure inside Forkhelpers.safe_close_and_exec.
+   Although the exact way of this reproduction is never supposed to
+   happen in the real world an internal failure could happen for instance
+   if forkexecd daemon is restarted for a moment so make sure we are
+   able to detect and handle these cases *)
+let test_internal_failure_error () =
+  let initial_fd_count = fd_count () in
+  let leak_fd_detect () =
+    let current_fd_count = fd_count () in
+    if current_fd_count <> initial_fd_count then
+      fail
```

Comment:
It's existing code but a function like `fail` would be easier to use if it would take printf-style arguments.

Reply:
Changed, I hope I got it right what you meant. In the meantime changed other minor things